pkg/controller/logstorage/elastic/elastic_controller.go

```diff
@@ -0,0 +1,725 @@
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating resource", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	if managementClusterConnection == nil {
+		if elasticsearch == nil || elasticsearch.Status.Phase != esv1.ElasticsearchReadyPhase {
+			r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for Elasticsearch cluster to be operational", nil, reqLogger)
+			return reconcile.Result{}, nil
+		}
+
+		if kibanaEnabled && (kibana == nil || kibana.Status.AssociationStatus != cmnv1.AssociationEstablished) {
+			r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for Kibana cluster to be operational", nil, reqLogger)
+			return reconcile.Result{}, nil
+		}
+
+		if !r.multiTenant {
+			// TODO: Need to handle this gracefully, since these will only succeed after ES is properly running.
+			// In multi-tenant mode, and probably single-tenant as well, this should be handled by someone other than the operator.
+			// Either out of band, or by Linseed.
+			result, _, err := r.applyILMPolicies(ls, reqLogger, ctx)
+			if err != nil {
+				return result, err
+			}
+		}
+
+		result, _, err := r.validateLogStorage(curatorSecrets, esLicenseType, reqLogger, ctx)
+		if err != nil {
+			return result, err
+		}
+	}
+
+	r.status.ReadyToMonitor()
+	r.status.ClearDegraded()
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ElasticSubController) validateLogStorage(curatorSecrets []*corev1.Secret, esLicenseType render.ElasticsearchLicenseType, reqLogger logr.Logger, ctx context.Context) (reconcile.Result, bool, error) {
+	var err error
+
+	if len(curatorSecrets) == 0 {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for curator secrets to become available", nil, reqLogger)
+		return reconcile.Result{}, false, nil
+	}
+
+	// kube-controller creates the ConfigMap and Secret needed for SSO into Kibana.
+	// If elastisearch uses basic license, degrade logstorage if the ConfigMap and Secret
+	// needed for logging user into Kibana is not available.
+	if esLicenseType == render.ElasticsearchLicenseTypeBasic {
+		if err = r.checkOIDCUsersEsResource(ctx); err != nil {
+			r.status.SetDegraded(operatorv1.ResourceReadError, "Failed to get oidc user Secret and ConfigMap", err, reqLogger)
+			return reconcile.Result{}, false, err
+		}
+	}
+	return reconcile.Result{}, true, nil
+}
+
+func (r *ElasticSubController) checkOIDCUsersEsResource(ctx context.Context) error {
+	if err := r.client.Get(ctx, types.NamespacedName{Name: render.OIDCUsersConfigMapName, Namespace: render.ElasticsearchNamespace}, &corev1.ConfigMap{}); err != nil {
+		return err
+	}
+
+	if err := r.client.Get(ctx, types.NamespacedName{Name: render.OIDCUsersEsSecreteName, Namespace: render.ElasticsearchNamespace}, &corev1.Secret{}); err != nil {
+		return err
+	}
+	return nil
+}
+
+// TODO: This shouldn't be done in the operator. Linseed should probably hanle this for single-tenant setups, and multi-tenant should
```

Comment:
For this PR, I have essentially commented this out for multi-tenant scenarios. I did this in order to work around the way credentials have changed in this PR - es-kube-controllers runs per-tenant, and installs secrets in per-tenant namespaces, and so right now the operator doesn't have credentials to talk to ES in this PR.

We have a few potential options for resolving this:

- Move operator secret provisioning out of es-kube-controllers, and into a cluster-scoped controller
- Make es-kube-controllers cluster-scoped instead of per-tenant
- Operator uses admin credentials for talking to ES, since it has those available already without any help.
- Remove ILM programming from the operator, move it to Linseed and/or another place.

I am partial to the latter.